test/jdk/jdk/lambda/separate/ClassToInterfaceConverter.java

```diff
@@ -23,75 +23,43 @@
-        ClassFile.CpUtf8 utf8 = (ClassFile.CpUtf8)entry;
-        if (v.length() != utf8.bytes.length) {
-            return false;
-        }
-        for (int i = 0; i < v.length(); ++i) {
-            if (v.charAt(i) != utf8.bytes[i]) {
-                return false;
-            }
-        }
-        return true;
-    }
-
-    private void convertToInterface(ClassFile cf) {
-        cf.access_flags = 0x0601; // ACC_INTERFACE | ACC_ABSTRACT | ACC_PUBLIC
-        ArrayList<ClassFile.Method> new_methods = new ArrayList<>();
-        // Find <init> method and delete it
-        for (int i = 0; i < cf.methods.size(); ++i) {
-            ClassFile.Method method = cf.methods.get(i);
-            ClassFile.CpEntry name = cf.constant_pool.get(method.name_index);
-            if (!utf8Matches(name, "<init>")) {
-                new_methods.add(method);
-            }
-        }
-        cf.methods = new_methods;
-        //  Convert method tag. Find Methodref, which is not "<init>" and only invoked by other methods
-        //  in the interface, convert it to InterfaceMethodref
-        ArrayList<ClassFile.CpEntry> cpool = new ArrayList<>();
-        for (int i = 0; i < cf.constant_pool.size(); i++) {
-            ClassFile.CpEntry ce = cf.constant_pool.get(i);
-            if (ce instanceof ClassFile.CpMethodRef) {
-                ClassFile.CpMethodRef me = (ClassFile.CpMethodRef)ce;
-                ClassFile.CpNameAndType nameType = (ClassFile.CpNameAndType)cf.constant_pool.get(me.name_and_type_index);
-                ClassFile.CpEntry name = cf.constant_pool.get(nameType.name_index);
-                if (!utf8Matches(name, "<init>") && cf.this_class == me.class_index) {
-                    ClassFile.CpInterfaceMethodRef newEntry = new ClassFile.CpInterfaceMethodRef();
-                    newEntry.class_index = me.class_index;
-                    newEntry.name_and_type_index = me.name_and_type_index;
-                    ce = newEntry;
-                }
+    private byte[] convertToInterface(ClassModel classModel) {
+        //  Convert method tag. Find Methodref which is only invoked by other methods
+        //  in the interface, convert it to InterfaceMethodref.  If opcode is invokevirtual,
+        //  convert it to invokeinterface
+        CodeTransform ct = (b, e) -> {
+            if (e instanceof InvokeInstruction i && i.owner() == classModel.thisClass()) {
+                Opcode opcode = i.opcode() == Opcode.INVOKEVIRTUAL ? Opcode.INVOKEINTERFACE : i.opcode();
+                b.invokeInstruction(opcode, i.owner().asSymbol(),
+                        i.name().stringValue(), i.typeSymbol(), true);
+            } else {
+                b.with(e);
             }
-            cpool.add(ce);
-        }
-        cf.constant_pool = cpool;
+        };
+        
+        return Classfile.of().transform(classModel,
+            ClassTransform.dropping(ce -> ce instanceof MethodModel mm && mm.methodName().stringValue().equals("<init>"))
```

Comment:
`Utf8Entry::equalsString(…)` might be more efficient:
```suggestion
            ClassTransform.dropping(ce -> ce instanceof MethodModel mm && mm.methodName().equalsString("<init>"))
```